endpoint.go

```diff
@@ -549,3 +554,48 @@ func (ep *endpoint) DataScope() datastore.DataScope {
 func (ep *endpoint) isLocalScoped() bool {
 	return ep.DataScope() == datastore.LocalScope
 }
+
+func (ep *endpoint) assignAddress() (err error) {
```

Comment:
Let's not use naked returns for this function. It's only to be used for 4-5 line functions.